core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java

```diff
@@ -839,6 +806,67 @@ RelNode changeTraitsUsingConverters(
     prunedNodes.add(rel);
   }
 
+
+  /** Returns whether to skip a match. This happens if any of the
+   * {@link RelNode}s have importance zero. */
+  boolean skipMatch(VolcanoRuleMatch match) {
```

Comment:
1. Currently, the planner is in charge of matching rules and pruning nodes (by importances or other mechanism). So I think it is more natural that planner judges whether a rule match is validate or not. 
2. By moving to Planner, it could be easier to overwrite the logic by users, adding customized pruning logic

Reply:
moved back